### PR TITLE
Create Database for Liked Tracks

### PR DESCRIPTION
**Pull request by thuannv-1**

### CODE REVIEW

File: SCPlayer/SCPlayer/Database/TrackEntity.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    static let shared = TrackEntity()
+    
+    private let tableTrack = Table("tableTrack")
+    
+    private let trackID = Expression<Int>("trackID")
+    
+    private init() {
+        do {
+            if let connection = Database.shared.connection {
+                try connection.run(tableTrack.create(temporary: false, ifNotExists: true, withoutRowid: false, block: { [unowned self] table in
```
Comment (datdt-1935) [inline SCPlayer/SCPlayer/Database/TrackEntity.swift]: break line

File: SCPlayer/SCPlayer/Database/TrackEntity.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        do {
+            let insert = tableTrack.insert(self.trackID <- trackID)
+            try Database.shared.connection?.run(insert)
+            print("Inserted Successfully !!!")
+        } catch {
+            print("Failed to insert: \(error)")
+        }
+    }
+    
+    func queryALl() -> AnySequence<Row>? {
```
Comment (datdt-1935) [inline SCPlayer/SCPlayer/Database/TrackEntity.swift]:
```suggestion
    func queryAll() -> AnySequence<Row>? {
```

File: SCPlayer/SCPlayer/Database/Database.swift [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    private init() {
+        self.database = createDatabase()
+    }
+    
+    func createDatabase() -> OpaquePointer? {
+        let path = "likeTrack.sqlite3"
+        var database: OpaquePointer?
+        
+        do {
+            let databasePath = try FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: false).appendingPathExtension(path)
```
Comment (datdt-1935) [inline SCPlayer/SCPlayer/Database/Database.swift]: break line

File: SCPlayer/SCPlayer/Database/Database.swift [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import SQLite3
+
+class TrackDatabase {
+    
+    static let shared = TrackDatabase()
+    
+    private var database: OpaquePointer?
+   
+    private init() {
+        self.database = createDatabase()
```
Comment (datdt-1935) [inline SCPlayer/SCPlayer/Database/Database.swift]: bỏ self

Comment (thuannv-1) [reply SCPlayer/SCPlayer/Database/Database.swift]: Dạ Vâng ạ

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**